### PR TITLE
Establish Engineering Team roles

### PR DESCRIPTION
**Pull request by carjug**
### What changed
Added "Engineering Team roles" section to `how_we_work.md`

@amymok @alexsoble I wasn't sure exactly what to put under the "Tech Lead role" subheading. Very open to suggestions there! Especially, would love your feedback @amymok since you're the Tech Lead after all :)

### Addresses
Linked issue #56 

### Preview link:
https://github.com/18F/OPRE-Unicorn/blob/0061fa980d01d800f278cb360c14465c945d8d91/docs/how_we_work.md

### CODE REVIEW

File: docs/how_we_work.md [hunk 1]
```diff
@@ -30,6 +30,29 @@
...
+
+### Tech Lead role
+
+Amy Mok (AM) will act as our Tech Lead for this project, with backup support from Alex Soble (ARS) and Carly Jugler (CJ).
+
+### Table of responsibilities
+
+Task/Responsibility | Primary | Backup
+-- | -- | --
+Keep tabs on the different eng streams and whether they’re flowing smoothly individually and together<br>
```
Comment (alexsoble) [inline docs/how_we_work.md:43]:
```suggestion
Keep tabs on the different eng streams and whether they’re flowing smoothly individually and together | AM | ARS
```

In the preview it looks like the first row of the table is blank; this suggestion would fill in that role.

**Comment by alexsoble** [comment]
@carjug In terms of what goes under the Tech Lead subheadding: I think traditionally a tech lead might own rows 1 through 5 of the table we sketched out; in our case we're distributing those responsibilities a little more broadly across the team. Does that help? 

Or put another way, I don't think we necessarily need to add things under the Tech Lead subhead that won't be in the table of tasks/responsibilities.

**Review by amymok — COMMENTED** [review]
This looks great!  Agree with Alex, we don't need to add anything else since we have clearly laid out the responsibilities.

**Review by alexsoble — APPROVED** [review]

**Review by ninamak — APPROVED** [review]
I still have concerns about dividing up the Tech Lead responsibilities, but I'm happy to give it a try.  Can we build in a retro in the near future to check in on how things are going with the role, how people are feeling about the load, etc?

**Comment by carjug** [comment]
@ninamak I think talking about the tech lead role in our regularly scheduled retro is totally reasonable!
It's worth mentioning that Alex, Amy and I feel that if one person were to take on all of the listed responsibilities, we would either need to be prepared to significantly cut scope for the product or ask for more money for another engineer.

**Comment by amymok** [comment]
@ninamak the way the engineers are in this project are more defined by our expertises which is a bit different from some of the other projects which started with one tech lead who owns the entire piece with knowledge in all the area from the start, and then building out the engineering team based on additional needs to do the work.

The tech lead in this project will still need to check in with each engineer to understand their domains and needs for this project, and then the lead can serve to understand how each stream is going based on that to align it for the overall project/goals.

**Comment by carjug** [comment]
@amymok if you approve can you merge, plz 🙏 💟

**Review by amymok — APPROVED** [review]
Sorry about the delay!  LGTM!

**Pull request merged**